packages/salesforcedx-vscode-core/src/commands/forceSourceDiff.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  DiffResultParser,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, hasRootWorkspace, OrgAuthInfo } from '../util';
+import {
+  FilePathGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+export class ForceSourceDiffExecutor extends SfdxCommandletExecutor<string> {
+  public build(filePath: string): Command {
+    const commandBuilder = new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_source_diff_text'))
+      .withArg('force:source:diff')
+      .withLogName('force_source_diff')
+      .withFlag('--sourcepath', filePath)
+      .withJson();
+    return commandBuilder.build();
+  }
+
+  public async execute(response: ContinueResponse<string>): Promise<void> {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath(),
+      env: { SFDX_JSON_TO_STDOUT: 'true' }
+    }).execute(cancellationToken);
+
+    channelService.streamCommandStartStop(execution);
+    channelService.showChannelOutput();
+
+    let stdOut = '';
+    execution.stdoutSubject.subscribe(realData => {
+      stdOut += realData.toString();
+    });
+
+    execution.processExitSubject.subscribe(async () => {
+      this.logMetric(execution.command.logName, startTime);
+      await handleDiffResponse(stdOut);
+    });
+
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+export async function handleDiffResponse(stdOut: string) {
+  try {
+    const diffParser = new DiffResultParser(stdOut);
+    const diffParserSuccess = diffParser.getSuccessResponse();
+    const diffParserError = diffParser.getErrorResponse();
+
+    if (diffParserSuccess) {
+      const diffResult = diffParserSuccess.result;
+      const remote = vscode.Uri.file(diffResult.remote);
```

Comment:
Updated this from `vscode.Uri.parse` to `vscode.Uri.file` since it was throwing the following error `Error: BAD uri lacks scheme, falling back to file-scheme.`. This was because I was providing a file path as a string and without a scheme e.g. `User/.../fileName.cls` instead of `file:///User/.../fileName.cls`.